client/src/pages/DashBoard.js

```diff
@@ -0,0 +1,484 @@
+          if (page === lastPage) {
+            dispatch(
+              postsActions.setLoadingAction({
+                isLoading: true,
+                loadMore: false,
+              }),
+            );
+          }
+          const loadedPosts = posts.reduce((obj, item) => {
+            obj[item._id] = item;
+            return obj;
+          }, {});
+          if (postsInState) {
+            dispatch(
+              postsActions.fetchPostsSuccess({
+                posts: { ...postsInState, ...loadedPosts },
+              }),
+            );
+          } else if (Object.keys(postsList).length && page) {
+            dispatch(
+              postsActions.fetchPostsSuccess({
+                posts: { ...postsList, ...loadedPosts },
+              }),
+            );
+          } else {
+            dispatch(
+              postsActions.fetchPostsSuccess({
+                posts: { ...loadedPosts },
+              }),
+            );
+          }
+        } else if (posts) {
+          dispatch(
+            postsActions.fetchPostsSuccess({
+              posts: { ...postsList },
+            }),
+          );
+          dispatch(postsActions.finishLoadingAction());
+        } else {
+          dispatch(postsActions.finishLoadingAction());
+        }
+      } catch (error) {
+        dispatch(postsActions.fetchPostsError(error));
+        dispatch(postsActions.finishLoadingAction());
+      }
+    }
+  };
+
+  useEffect(() => {
+    getStateFromQuery();
+  }, [history.location.search]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    setQueryFromState();
+  }, [applyFilters, status]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    refetchPosts(); // will trigger loadPosts(if needed) (by toggling toggleRefetch)
+  }, [queryParams]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (applyFilters) {
+      loadPosts();
+    }
+  }, [toggleRefetch, page]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const isItemLoaded = useCallback((index) => !!feedPosts[index], [feedPosts]);
+
+  const loadNextPage = useCallback(
+    ({ stopIndex, loadLogsPage }) => {
+      if (loadLogsPage !== undefined) {
+        dispatch(postsActions.setPageAction(loadLogsPage));
+        return Promise.resolve();
+      }
+      if (
+        !isLoading &&
+        loadMore &&
+        stopIndex >= feedPosts.length &&
+        feedPosts.length
+      ) {
+        return new Promise((resolve) => {
+          dispatch(postsActions.setNextPageAction());
+          dispatchAction(SET_VALUE, "applyFilters", true);
+          resolve();
+        });
+      } else {
+        return Promise.resolve();
+      }
+    },
+    [feedPosts.length, isLoading, loadMore], // eslint-disable-line react-hooks/exhaustive-deps
+  );
+
+  useEffect(() => {
+    setItemCount(loadMore ? feedPosts.length + 1 : feedPosts.length);
+  }, [feedPosts.length, loadMore]);
```

Comment:
miss one? `// eslint-disable-line react-hooks/exhaustive-deps`

Reply:
no, it doesn't need that. Dependency array is already included all dependants.